Turn off watchdog when done in tests

khwilliamson opened the pull request:
If you set a watchdog timer, you should clear it when its no longer needed.  Otherwise it can go off, aborting your test script.

In these more than two dozen test files, it hasn't mostly been a problem because the script finishes before the timer goes off.  But I bet that some heisenbugs have been the result of not clearing it.


* This set of changes does not require a perldelta entry.

tonycoz commented on dist/IO/t/io_multihomed.t:
```diff
@@ -117,3 +117,5 @@ if (my $pid = fork()) {
     die;
 }
+
+watchdog(o);
```
`watchdog(0)` ?

tonycoz commented:
The test.pl currently shipped with IO (and I think threads, threads-shared) don't treat `$timeout == 0` specially.

khwilliamson committed:
Turn off watchdog when done in tests

If you set a watchdog timer, you should clear it when its no longer
needed.  Otherwise it can go off, aborting your test script.

In these twenty test files, it hasn't mostly been a problem because the
script finishes before the timer goes off.  But I bet that some
heisenbugs have been the result of not clearing it.

It actually shouldn't be necessary to clear the timer when that is the
final statement in the test file, but we may want to add a porting test
that makes sure watchdogs are cleared, and always adding a clear enables
such a test to properly work.

khwilliamson commented:
For this p.r., I removed the changes to IO and threads
